src/PrestaShopBundle/Service/Grid/ResponseBuilder.php

```diff
@@ -99,4 +120,57 @@ public function buildSearchResponse(
 
         return new RedirectResponse($redirectUrl, 302);
     }
+
+    /**
+     * @param array $filtersFormData
+     * @param string $filterId
+     *
+     * @return array
+     */
+    private function checkFiltersFormData(array $filtersFormData, string $filterId): array
```

Comment:
I'm not sure you need to transform the input array, since the generator will ignore `null` fields any way
Maybe this function should only check if the array is empty, reset the filter if necessary and return nothing (or void)

And rename this `resetPersistedFilters`, then even the check doesn't need to be in this method, you would just need to do:

```php
if ($filtersForm->isSubmitted()) {
    if ($this->checkIsFormDataEmpty($filtersForm->getData())) {
        $this->resetPersistedFilter($filterId);
    }

    $redirectParams = [
        $filterId => [
            'filters' => $filtersForm->getData(),
        ]
    ];
}
```